tests/test_build_public_bom.py

```diff
@@ -286,6 +310,45 @@ def test_delete_last_nested_components(self) -> None:
         public_component = b_p_b.remove_component_tagged_internal(component, validator)
         self.assertEqual(expected_component, public_component[1])
 
+    def test_delete_internal_properties(self) -> None:
+        component = {
+            "name": "test",
+            "properties": [
+                {"name": "internal:stuff", "value": "should be gone"},
+                {"name": "stuff", "value": "still there"},
+            ],
+        }
+        b_p_b.remove_internal_information_from_properties(component)
+        expected = {
+            "name": "test",
+            "properties": [{"name": "stuff", "value": "still there"}],
+        }
+        self.assertEqual(component, expected)
+
+    def test_not_delete_internal_properties(self) -> None:
+        component = {
+            "name": "test",
+            "properties": [
+                {"name": "stuff:internal", "value": "still there"},
+                {"name": "stuff", "value": "still there"},
+            ],
+        }
+        expected = component.copy()
```

Comment:
Careful, this is a subtle bug. [`dict.copy()`](https://docs.python.org/3/library/stdtypes.html#dict.copy) creates shallow copies. Use [`copy.deepcopy()`](https://docs.python.org/3/library/copy.html#copy.deepcopy), instead.

Easy to prove:
```python
original = {"value": "foo", "list": [1]}
copy = original.copy()
assert original == copy # good, copy and original are identical

copy["list"].append(2) # modify the copy
assert original == copy # ouch, should raise AssertionError but doesn't.
```

In consequence, whatever the function under test does to any nested list or dict inside `component` would also be reflected on `expected`. The assertion would always be true.